x-pack/plugins/observability_solution/apm/public/components/routing/templates/apm_main_template/index.tsx

```diff
@@ -77,8 +78,18 @@ export function ApmMainTemplate({
     entityCentricExperience,
     true
   );
-  const { isEntityCentricExperienceViewEnabled, serviceInventoryViewLocalStorageSetting } =
-    useEntityManagerEnablementContext();
+
+  const {
+    isEntityCentricExperienceViewEnabled,
+    serviceInventoryViewLocalStorageSetting,
+    setServiceInventoryViewLocalStorageSetting,
+  } = useEntityManagerEnablementContext();
+
+  useEffect(() => {
+    if (serviceInventoryViewLocalStorageSetting === ServiceInventoryView.entity) {
+      setServiceInventoryViewLocalStorageSetting(ServiceInventoryView.classic);
```

Comment:
@cauemarcondes, I've implemented a temporary fix to address this [comment](https://github.com/elastic/kibana/pull/193997#issuecomment-2374122587). It was necessary to apply this change after the component mounts, as the context wasn’t rendered yet, leading to runtime errors.

I also noticed that @kpatticha created an [issue](https://github.com/elastic/kibana/issues/194114) related to the cleanup of the services inventory view context. Once that task is completed, we can remove this temporary fix.

Reply:
What do you think of just removing [this](https://github.com/elastic/kibana/blob/main/x-pack/plugins/observability_solution/apm/public/components/app/service_inventory/index.tsx#L39-L43)  condition and always load the <ApmServiceInventory />

Reply:
If that’s the only part that could be affected, let’s proceed with that approach. I was concerned there might be other areas in APM relying on it for some reason.

I'll also remove the check we have for the `noDataConfig` prop for `ObservabilityPageTemplate` in the `ApmMainTemplate`.

Thanks for the suggestion!

Reply:
I think if you also remove this check https://github.com/elastic/kibana/blob/main/x-pack/plugins/observability_solution/apm/public/context/entity_manager_context/entity_manager_context.tsx#L67 we should be good to go.